certora/specs/SocialRecoveryModule.spec

```
@@ -1,26 +1,160 @@
 }
 
-// This is a dummy rule to verify the Safe Contract Setup with the Social Recovery
-// Module is working as intended.
-rule recoveryModuleCanBeDisabled {
-    env e;
-    address prevModule;
+// This integrity rule verifies that the addGuardianWithThreshold(...) if executes ensures that
+// the Social Recovery Module is enabled, the caller to the Module has to be the Safe Contract,
+// the new guardian is added to the guardian list, and no other account (guardian or not) is affected.
+rule addGuardianWorksAsExpected(env e, address guardian, uint256 threshold, address otherAccount) {
+
+    uint256 currentGuardiansCount = guardianStorageContract.entries[safeContract].count;
+
+    require guardianStorageContract.entries[safeContract].count == guardianStorageContract.countGuardians(safeContract);
+
+    require guardian != otherAccount;
+    bool otherAccountIsGuardian = currentContract.isGuardian(safeContract, otherAccount);
+
+    currentContract.addGuardianWithThreshold(e, safeContract, guardian, threshold);
+
+    assert safeContract.isModuleEnabled(currentContract);
+    assert e.msg.sender == safeContract;
+    assert currentContract.isGuardian(safeContract, guardian);
+    assert otherAccountIsGuardian == currentContract.isGuardian(safeContract, otherAccount);
+    assert currentGuardiansCount + 1 == to_mathint(guardianStorageContract.entries[safeContract].count);
+}
+
+// This integrity rule verifies that the guardian can always be added considering ideal conditions.
+rule guardianCanAlwaysBeAdded(env e, address guardian, uint256 threshold) {
+
+    requireSocialRecoveryModuleEnabled();
+
+    require e.msg.value == 0;
+    require threshold > 0;
+    require guardianStorageContract.entries[safeContract].count < max_uint256; // To prevent overflow (Realistically can't reach).
+    require guardianStorageContract.entries[safeContract].count == guardianStorageContract.countGuardians(safeContract);
+
+    require guardian != 0;
+    require guardian != SENTINEL();
+    require guardian != safeContract;
+
+    require !safeContract.isOwner(guardian);
+    require !currentContract.isGuardian(safeContract, guardian);
+
+    require threshold < guardianStorageContract.countGuardians(safeContract);
+
+    require e.msg.sender == safeContract;
+    currentContract.addGuardianWithThreshold@withrevert(e, safeContract, guardian, threshold);
+    bool isReverted = lastReverted;
+
+    assert !isReverted && currentContract.isGuardian(safeContract, guardian);
+}
+
+// This integrity rule verifies that the addition of a new guardian always reverts if the guardian is already added.
+rule addGuardiansRevertIfDuplicateGuardian(env e, address guardian, uint256 threshold) {
+
+    require currentContract.isGuardian(safeContract, guardian);
+
+    currentContract.addGuardianWithThreshold@withrevert(e, safeContract, guardian, threshold);
+    bool isReverted = lastReverted;
+
+    assert lastReverted;
```

Nit: to avoid requires:

```suggestion
    bool wasGuardian = currentContract.isGuardian(safeContract, guardian);

    currentContract.addGuardianWithThreshold@withrevert(e, safeContract, guardian, threshold);
    bool isReverted = lastReverted;

    assert wasGuardian => isReverted;
```

---

Done 👍🏾